chainer/functions/split_axis.py

```diff
@@ -33,21 +32,20 @@ def forward_gpu(self, x):
         self.cdimx = xshape[self.axis]
         self.rdim = numpy.prod(xshape[self.axis + 1:])
 
-        indices = self.indices_or_sections
-        if isinstance(indices, collections.Iterable):
-            indices = list(indices)
-            indices.append(xshape[self.axis])
+        if isinstance(self.indices_or_sections, collections.Iterable):
+            ind = list(self.indices_or_sections)
+            ind.append(xshape[self.axis])
         else:
-            if xshape[self.axis] % indices:
+            sec = self.indices_or_sections
```

Comment:
It is better to check if `self.dices_or_sections` is integer and raise an error if this assumption is not satisfied.